specs/vep-554-apache-airflow-integration/README.MD

```diff
@@ -0,0 +1,401 @@
+  them should be executed.
+* Trigger a job based on previous job status in the DAG - when there is "Data Mart tables
+  dependencies (DAG)", some jobs should be available as optional (e.g. DAG should not
+  fail if steps fail).
+
+#### Contribute a custom provider to Airflow Community
+
+#### Executing Data Jobs should be enabled by private Airflow deployments
+
+User cases:
+
+* Users want to create operational and analytics pipelines of jobs.
+* Users already own their Airflow deployment.
+
+#### Comparison of VDK with Apache Airflow
+
+### Non-goals
+
+#### Provide managed Apache Airflow deployment
+
+## High-level design
+
+![high-level-diagram.png](high-level-diagram.png)
+
+Apache Airflow 2 is built in a modular way. The "Core" of Apache Airflow provides core
+scheduler functionality which allows you to write some basic tasks, but the capabilities
+of Apache Airflow can be extended by installing additional packages, called providers.
+
+Providers can contain operators, sensors, hooks, and transfer operators to communicate
+with a multitude of external systems, but they can also extend Airflow core with new
+capabilities.
+
+The proposed design describes the solution for supporting multi-jobs non-linear analytics
+by integrating VDK with Apache Airflow. For this purpose, an Apache Airflow Provider
+called "VDK Provider" will be implemented and responsible for managing Data Jobs. It will
+give us the ability to define DAGs in which the tasks (jobs) will be defined by
+instantiating custom operators. The operators will start and monitor Data Job executions
+via the Control Service Execution API.
+
+In order to speed up the VDK popularization and provide easier integration with Airflow
+to our users, the VDK Provider will be contributed to the Airflow Community.
+
+## API design
+
+No changes to the public API.
+
+## Detailed design
+
+### VDK Provider
+
+The integration of VDK with Apache Airflow will be accomplished by VDK Provider
+implementation. It will provide an ability to the users to define Airflow Workflow that
+consists of Data Jobs. Basically, it will be responsible for managing Data Job
+executions (starting, canceling, retrieving). As can be seen from the below diagram the
+provider will consist of a hook, operator, and sensor.
+
+![class-diagram.png](class-diagram.png)
+
+Users will be able to choose from two operating modes - synchronous and asynchronous,
+depending on their Data Jobs and needs. The synchronous mode will be achieved by
+providing an operator derived from `airflow.models.baseoperator.BaseOperator`, which
+executes a Data Job and waits for its completion. The asynchronous mode will be achieved
+by providing a sensor operator derived from `airflow.sensors.base.BaseSensorOperator`,
+which polls on the specific time intervals for job execution completion and does not hold
+the Airflow Worker.
+
+* Availability.
+    * The availability of the VDK Provider will be managed by Airflow, since it is going
```

Comment:
In terms of availability our provider depends on Airflow but also on Control Service and Authorization server (per high-level-diagram). 

If airflow is down, it will not be available . 

If Control Service API or Auth Server is not working - would there be any re-tries? how it would handle networking issues (they tend to me intermittent) . What if we have high latency of the requests or bandwidth is low?